packages/core/src/login/webview/vue/amazonq/backend_amazonq.ts

```diff
@@ -19,11 +19,43 @@ export class AmazonQLoginWebview extends CommonAuthWebview {
         super(AmazonQLoginWebview.sourcePath)
     }
 
-    fetchConnection(): SsoConnection | undefined {
-        if (AuthUtil.instance.isConnected() && AuthUtil.instance.conn?.type === 'sso') {
-            return AuthUtil.instance.conn
+    async fetchConnections(): Promise<SsoConnection[] | undefined> {
+        const toolkitExt = vscode.extensions.getExtension('amazonwebservices.aws-toolkit-vscode')
+        const importedApi = toolkitExt?.exports
+        const connections: SsoConnection[] = []
+        const toolkitConnections = await importedApi?.listConnections()
+        toolkitConnections.forEach((connection: SsoConnection) => {
+            if (connection.scopes?.includes('codewhisperer:completions')) {
+                connections.push(connection)
+            } else {
+                connections.push({
+                    id: connection.id,
+                    label: connection.label,
+                    scopes: amazonQScopes,
```

Comment:
how do we know the connection has these scopes?

Reply:
It does not. 


If user attempt to use Q,  user will be asked to redo the AuthZ in browser